apps/emqx_resource/src/emqx_resource_buffer_worker.erl

```diff
@@ -295,19 +295,19 @@ pick_call(Id, Key, Query, Timeout) ->
     ?PICK(Id, Key, Pid, begin
         Caller = self(),
         MRef = erlang:monitor(process, Pid, [{alias, reply_demonitor}]),
-        From = {Caller, MRef},
+        From = {Caller, [alias | MRef]},
```

Comment:
This function is pretty complicated, it would be great to explain what's going on in a comment.
Q: why does From contain an improper list?

Reply:
I'll drop the usage of `gen_statem:reply` and go back to my first solution of just sending to `MRef` the reply, after Stone's comment above.

Reply:
:heavy_check_mark: